### ISSUE TITLE
Cannot run Makefile

### ISSUE THREAD

**Issue by shreyas-7** [astra-toolbox/astra-toolbox#152]
I am getting the following error while running on Mac OS 10.13, Xcode 9.3, Matlab_R2015b

This error comes when i run the Makefile.

/Applications/MATLAB_R2015b.app/bin/mex LDFLAGS="\$LDFLAGS  -g -fopenmp -L/Users/shreyas/.Trash/astra-1.8/build/linux/.libs -Wl,-rpath,'\$ORIGIN'" -cxx  -lpthread -lut -lastra -output matlab/mex/astra_mex_algorithm_c matlab/mex/astra_mex_algorithm_c.o matlab/mex/mexHelpFunctions.o matlab/mex/mexCopyDataHelpFunctions.o matlab/mex/mexInitFunctions.o matlab/mex/mexDataManagerHelpFunctions.o
Building with 'Xcode Clang++'.
clang: error: unsupported option '-fopenmp'

**Comment by wjp** [comment]
Could you try the git version? There are some changes for macOS support in there.

**Comment by shreyas-7** [comment]
It worked, thank you.

A small follow-up, on running astra_test after including the files via pathtool, I get the following error -

astra_test
Invalid MEX-file
'/Users/shreyas/Downloads/astra-toolbox-master-2/build/linux/matlab/mex/astra_mex_c.mexmaci64':
dlopen(/Users/shreyas/Downloads/astra-toolbox-master-2/build/linux/matlab/mex/astra_mex_c.mexmaci64,
6): Library not loaded: @loader_path/libastra.0.dylib
  Referenced from:
  /Users/shreyas/Downloads/astra-toolbox-master-2/build/linux/matlab/mex/astra_mex_c.mexmaci64
    Reason: image not found

Error in astra_mex (line 22)
    [varargout{:}] = astra_mex_c(varargin{:});

Error in astra_test (line 14)
if astra_mex('use_cuda')

**Comment by wjp** [comment]
Great to hear. Did you run `make install`? If you prefer not to, you'll have to put the `.mexmaci64` files and `libastra.0.dylib` in the same directory manually.

**Comment by shreyas-7** [comment]
Yep, I did run make install earlier as well, I ran it once again and
This is the log that I got -

$ make install
/bin/sh ./install-sh -m 755 -d /Users/shreyas/astra/matlab/mex
/bin/sh ./install-sh -m 755 /Users/shreyas/Downloads/astra-toolbox-master/build/linux/.libs/libastra.0.dylib /Users/shreyas/astra/matlab/mex
/bin/sh ./install-sh -m 755 -d /Users/shreyas/astra/matlab
/bin/sh ./install-sh -m 755 -d /Users/shreyas/astra/matlab/mex
/bin/sh ./install-sh -m 755 -d /Users/shreyas/astra/matlab/tools
/bin/sh ./install-sh -m 755 matlab/mex/astra_mex_algorithm_c.mexmaci64 matlab/mex/astra_mex_data2d_c.mexmaci64 matlab/mex/astra_mex_c.mexmaci64 matlab/mex/astra_mex_matrix_c.mexmaci64 matlab/mex/astra_mex_projector_c.mexmaci64 matlab/mex/astra_mex_projector3d_c.mexmaci64 matlab/mex/astra_mex_log_c.mexmaci64 matlab/mex/astra_mex_data3d_c.mexmaci64 matlab/mex/astra_mex_direct_c.mexmaci64 /Users/shreyas/astra/matlab/mex
/bin/sh ./install-sh -m 644 ./../../matlab/tools/*.m /Users/shreyas/astra/matlab/tools

I also did 
sudo find / -name "libastra.0.dylib"
which returns the following files 

/Users/shreyas/astra/matlab/mex/libastra.0.dylib
/Users/shreyas/Downloads/astra-toolbox-master/build/linux/.libs/libastra.0.dylib


I found that in first directory .mexmaci64 files were present, and they weren't in the second directory.
I copied the files from the first directory to the second one and now it works.

Thank you for your help.

**Comment by wjp** [comment]
Glad that it works now. Out of curiosity, which path did you add to your matlab path? `/Users/shreyas/astra/matlab/mex` or `/Users/shreyas/Downloads/astra-toolbox-master/build/linux/...` ?

**Comment by shreyas-7** [comment]
/Users/shreyas/Downlaods/astra-toolbox-master and all it's subfolders\
Do I need to add the other one as well?

Also, just minor query, when I run one of the samples like for example 
 --> s013_constraints

I get the following error

    Error using astra_mex_algorithm_c
    Unknown Algorithm.
    
    Error in astra_mex_algorithm (line 22)
        [varargout{:}] = astra_mex_algorithm_c(varargin{:});
    
    Error in astra_create_sino_gpu (line 44)
    alg_id = astra_mex_algorithm('create', cfg);
    
    Error in s013_constraints (line 19)
    [sinogram_id, sinogram] = astra_create_sino_gpu(P, proj_geom, vol_geom);


So it doesn't work because I don't have a GPU.
Is there an alternative algorithm for astra_create_sino_gpu that works on CPU? I tried astra_create_sino() but it works on only 2D images, and when I run astra_create_sino_3d_cuda() it says geometry class is not initialised ( I don't have CUDA)

So in conclusion, will I be able to run only 2D functionalities if I don't have a GPU? 

Thank You.

**Comment by wjp** [comment]
The other folder (`/Users/shreyas/astra/matlab`) is where the files were installed after building them. In theory adding that folder after running `make install` should be enough (without having to manually copy more files.)

You are correct that astra currently doesn't have 3D CPU functionality.

**Comment by shreyas-7** [comment]
Thanks for the reply.

I created a sinogram using the radon function of Matlab, is there a way I can reconstruct the image from this sinogram using the reconstruction algorithms of astra?

Thank you.

**Comment by da62b202** [comment]
Hi, 
I have problem as following, tested on 2D SART, worked on Windows 10, MATLAB R2017a:
Error in astra_mex_algorithm

    [varargout{:}] = astra_mex_algorithm_c(varargin{:});

So that the following could not initialize:

sart_id = astra_mex_algorithm('create', cfg);

**Comment by wjp** [comment]
Please open a new issue instead of responding to an old issue. Thank you.
